### PR TITLE
numpy parity: financial functions #128

### PR DESCRIPTION
**Pull request by dreamquster**
complement irr/mirr/nper/rate

### CODE REVIEW

File: math/src/main/scala/breeze/numerics/financial/package.scala [hunk 1]
```diff
@@ -97,4 +97,119 @@ package object financial {
   def principalPayments(rate: Double, numPeriods: Int, presentValue: Double, futureValue: Double = 0.0, when: PaymentTime = End): DenseVector[Double] = principalInterest(rate, numPeriods, presentValue, futureValue, when)._2
 
   def principalRemaining(rate: Double, numPeriods: Int, presentValue: Double, futureValue: Double = 0.0, when: PaymentTime = End): DenseVector[Double] = principalInterest(rate, numPeriods, presentValue, futureValue, when)._3
+
+
+
+  private def roots(coeffs: DenseVector[Double]) = {
+    val coeffsArray = coeffs.toArray;
+    val trailingZeros = coeffsArray.indexWhere(x =>
+      if (0 != x) true else false);
```
Comment (dlwh) [inline math/src/main/scala/breeze/numerics/financial/package.scala]: why not  just indexWhere(0 != _) ?

File: math/src/main/scala/breeze/numerics/financial/package.scala [hunk 2]
```diff
@@ -97,4 +97,119 @@ package object financial {
...
+      Option[Double](rates(argmin(abs(rates))))
+    }
+
+    rate
+  }
+
+  def modifiedInternalRateReturn(values:DenseVector[Double], financeRate:Double, reinvestRate:Double = 0) = {
+    val n = values.length
+    val positives = values.mapValues(x => if (0 < x) x else 0)
+    val posCnt = values.foldLeft(0.0)((cnt, x) => if (0 < x) (cnt + 1) else cnt)
```
Comment (dlwh) [inline math/src/main/scala/breeze/numerics/financial/package.scala]: why not just values.valuesIterator.count(_ > 0) ?

**Comment by dlwh** [comment]
thanks for doing this! just a few minor-ish suggestions

**Comment by dreamquster** [comment]
Thank for your instructment.  I'm a newer to Scala/breeze and not familiar with some iterator operation.
:)

**Pull request merged**